graphql-server/src/schemas/queries/samples.ts

```diff
@@ -371,23 +372,118 @@ export async function queryDashboardSamples({
     SKIP ${offset}
     LIMIT ${limit}
   `;
+}
 
+export async function queryDashboardSamples({
+  samplesCypherQuery,
+  oncotreeCache,
+}: {
+  samplesCypherQuery: string;
+  oncotreeCache: OncotreeCache | undefined;
+}) {
   const session = neo4jDriver.session();
   try {
-    const result = await session.run(cypherQuery);
-
+    const result = await session.run(samplesCypherQuery);
     return result.records.map((record) => {
       const recordObject = record.toObject().resultz;
-      const otCache = recordObject.oncotreeCode
-        ? (oncotreeCache.get(recordObject.oncotreeCode) as CachedOncotreeData)
-        : null;
       return {
         ...recordObject,
-        cancerType: otCache?.mainType,
-        cancerTypeDetailed: otCache?.name,
+        cancerType: oncotreeCache?.[recordObject.oncotreeCode]?.mainType,
+        cancerTypeDetailed: oncotreeCache?.[recordObject.oncotreeCode]?.name,
       };
     });
   } catch (error) {
     console.error("Error with queryDashboardSamples:", error);
   }
 }
+
+export function getAddlOtCodesMatchingCtOrCtdVals({
+  searchVals,
+  oncotreeCache,
+}: {
+  searchVals: QueryDashboardSamplesArgs["searchVals"];
+  oncotreeCache: OncotreeCache;
+}) {
+  let addlOncotreeCodes: Set<string> = new Set();
+  if (searchVals?.length) {
+    for (const [code, { name, mainType }] of Object.entries(oncotreeCache)) {
+      for (const val of searchVals) {
+        if (
+          name?.toLowerCase().includes(val?.toLowerCase()) ||
+          mainType?.toLowerCase().includes(val?.toLowerCase())
+        ) {
+          addlOncotreeCodes.add(code);
+        }
+      }
+    }
+  }
+  return Array.from(addlOncotreeCodes);
+}
+
+export type SampleDataForCacheUpdate = Record<
+  string,
+  Pick<
+    DashboardSample,
+    | "primaryId"
+    | "cmoSampleName"
+    | "historicalCmoSampleNames"
```

Comment:
good catch! I forgot about having to include this field in the query for latest updates.